front-end/src/hooks/voteContext.js

```diff
@@ -1,31 +1,57 @@
-import React, { useState, useContext, createContext } from 'react';
-import PropTypes from 'prop-types';
+import React, { useEffect, useState, useContext, createContext } from "react";
+import PropTypes from "prop-types";
+
+import { ConsumeAuth } from "./authContext";
+
+import {
+  setProposalsStorage,
+  getProposalsStorage,
+} from "./storage";
 
 const voteContext = createContext();
 
 function useProvideVote() {
-    const [proposals, setProposals] = useState([]);
+  const authContext = ConsumeAuth();
+
+  const [proposals, setProposals] = useState([]);
 
-    return {
-        proposals,
-        setProposals,
-    };
+  const addProposal = async (proposal) => {
+    const proposalExists = proposals.find(x => x.proposalId === proposal.proposalId)
+    if (!proposalExists) {
+      const newProposals = [...proposals, proposal]
+      setProposals(newProposals)
+      setProposalsStorage(newProposals)
+    }
+  }
+
+  useEffect(() => {
+    if (authContext.isLoggedIn()) {
+      const proposalsStorage = getProposalsStorage();
+      if (proposalsStorage) {
+        setProposals(proposalsStorage);
+      }
+    }
+  }, [authContext.isLoggedInValue]);

```

Comment:
authContext.isLoggedIn should be this maybe?

Reply:
authContext.isLoggedIn is a function, not a state